ui/docs/reference/integrations/github/checks.mdx

```diff
@@ -0,0 +1,44 @@
+# Taskcluster-Github Checks
+
+Taskcluster-Github supports [Github's Checks API](https://developer.github.com/v3/checks/).
+At the moment, the feature is in beta. If you would like to try it out:
+* Make sure you use `.taskcluster.yml` v1
+* Add `reporting: checks-v1` line at the root level of your `.taskcluster.yml`
+* You will need `queue:route:checks` scope for your `repo:github.com/org/repo:<action>` role.
+* If you are using a decision task, make sure you add `route.checks` to the `routes` of your tasks
+
+## Custom Output in Checks
+
+If you need to add some sort of custom output to your checks (for example, you would like to process
+your logs and add some results to the check's body):
+* You need to arrange for an artifact with the custom text (plain or Markdown) you'd like to add to your check 
+ (for the above example, you arrange the processing of your logs in the way you want, 
+ format the result the way you want using Markdown and save the result in an artifact)
+* Add something like 
+    ```YML
+    extra:
+        github:
+            customCheckRun:
+                textArtifactName: <artifact_name>
+    ```
+ to the definition of the task for which you want customize the check. If you don't want to bother with that,
+ you can just make sure you name your artifact `customCheckRunText.md`, and tc-gh would fetch that automatically.
+ Please note that at the moment we do not support private artifacts for this functionality, so do make sure
+ to add `public/` prefix to your artifact name when using `Queue.createArtifact` endpoint
+ (like this: `public/<file_name>`).
```

Comment:
The user probably won't be calling `createArtifact` themselves -- they'll be defining this as an artifact that the worker will upload for them.  So mentioning `createArtifact` here might confuse users.

Reply:
when I was writing this, I was thinking more of the admin type of audience(such as Tom here)who would actually be defining scripts and tasks. Those also might bethe developers themselves.

I'm not sure what's the best way of phrasing this, do you have ideas?

Reply:
Actually, your idea is expressed in the previous comment, I think.

Reply:
It might be better to write it in a less instructional style (since instructional style is fairly specific to the audience) and more of a reference style.  Something like

When a task completes, the Taskcluster-GitHub service can optionally include task-specific information in the check run displayed in the GitHub checks tab.  To do so, it looks for an artifact on the task named `public/github/customCheckRunText.md` and uses the artifact content as the check run text.  The artifact name can be customized by setting `task.extra.github.customCheckRun.textArtifactName` in the task definition.  For example:
```
payload:
  ...
  artifacts:
    public/github/customCheckRunText.md:
      type: file
      path: checkrun.md
```
or, if for example all build results are in `public/results`:
```
payload:
  ...
  artifacts:
    public/results:
      type: directory
      path: results
task:
  extra:
    github:
      customCheckRun:
        textArtifactName: public/results/checkrun.md
```